jmespath/visitor.py

```diff
@@ -29,9 +29,9 @@ def _is_special_integer_case(x, y):
     # Also need to consider that:
     # >>> 0 in [True, False]
     # True
-    if x is 0 or x is 1:
+    if type(x) is int and (x == 0 or x == 1):
```

Comment:
```suggestion
    if isinstance(x, int) and (x == 0 or x == 1):
```
PEP 8

Reply:
No, as you can see from the comment preceding the code, or by running the tests, we actually *require* the behavior that PEP 8 is trying to guard against.

```console
>>> type(True) is int
False
>>> isinstance(True, int)
True
```

(This is at least a defined part of the language standard where the behavior of `is 0` and `is 1` is not.)